src/components/Header/Header.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useContext, useState, useEffect } from 'react';
+import UserContext from '../../Context/UserContext';
+import FilterableList from '../FilterableList/FilterableList';
+import Button from '../Button/Button';
+import MenuList from '../MenuList/MenuList';
+import styles from './Header.scss';
+import axios from 'axios';
+import { useHistory } from 'react-router';
+
+function Header() {
+  const [isVisible, setIsVisible] = useState(false);
+  const [cities, setCities] = useState([]);
+  const [inputValue, setInputValue] = useState('');
+  const history = useHistory();
+
+  const handleClick = () => {
+    setIsVisible(prev => !prev);
+  };
+
+  const context = useContext(UserContext);
+  const token = context.token;
+
+  useEffect(() => {
+    const cancelTokenSource = axios.CancelToken.source();
+    if (token !== '' && inputValue !== '') {
```

Comment:
Token related logic is not something that should be put in each request performer. Try to resolve token validity on one centralised place. IF token is expired or about to expire - refresh it.

Reply:
I will remove Context and ContextProvider(in current task about Redux) and token will be store in Redex Store.